modules/snmp/collect.go

```diff
@@ -38,7 +38,13 @@ func (s *SNMP) collectChart(collected map[string]int64, OIDs []string) error {
 	}
 
 	for i, oid := range OIDs {
-		collected[oid] = gosnmp.ToBigInt(result.Variables[i].Value).Int64()
+		switch result.Variables[i].Type {
+		case gosnmp.NoSuchInstance:
+		case gosnmp.NoSuchObject:
+			continue
+		default:
+			collected[oid] = gosnmp.ToBigInt(result.Variables[i].Value).Int64()
+		}
```

Comment:
```suggestion
		switch result.Variables[i].Type {
		case gosnmp.NoSuchInstance:
		case gosnmp.NoSuchObject:
		default:
			collected[oid] = gosnmp.ToBigInt(result.Variables[i].Value).Int64()
		}
```

We don't really need `continue` here

---

@codeguru1 can we have a test case for different types?

Reply:
I'm working on test cases